### ISSUE TITLE
Parameterized constructor

### ISSUE THREAD

**Issue by Udayv56** [Udayv56/CPP#17]
**CODE-**
#include <iostream>
#include<cmath>
using namespace std;
//constructor 
class Point{
    int x,y;
    friend int disPoint(Point p1, Point p2);
  public:
    Point(int ,int); 
    void printNumber(void){
        cout<<"Your point  is ("<<x<<" , "<<y<<")"<<endl;
    }
    
};
Point:: Point(int a,int b){  //paramiterized constructor
    x=a;
    y=b;
}

int disPoint(Point p1, Point p2){
    return (p2.x -p1.x)*(p2.x -p1.x) + (p2.y -p1.y)*(p2.y -p1.y);
}
int main()
{
    Point p1(2,3); //implicit call
    Point p2 = Point(6,7);
    p1.printNumber();
    p2.printNumber();
    int res = disPoint(p1, p2);
    cout<<"The distance between p1 and p2 is "<<sqrt(res)<<endl;
    return 0;
}

**OUTPUT-**
In the provided code, a parameterized constructor is used in the `Point` class. A parameterized constructor is a constructor that takes one or more parameters to initialize the object's data members during object creation.

In the code, the parameterized constructor for the `Point` class is defined as `Point(int a, int b);`. It takes two integer parameters `a` and `b`, which will be used to initialize the `x` and `y` data members of the `Point` class.

Inside the constructor definition, the values of `a` and `b` are assigned to `x` and `y`, respectively, using the assignment statements `x = a;` and `y = b;`.

In the `main()` function, two objects `p1` and `p2` of the `Point` class are created using the parameterized constructor. The object `p1` is created using implicit constructor call syntax, where the values `2` and `3` are passed as arguments to the constructor. The object `p2` is created explicitly using the constructor call syntax, where `Point(6, 7)` is used to construct the object.

The `printNumber()` member function of the `Point` class is then called on both `p1` and `p2` objects to print their respective `x` and `y` values.

Furthermore, there is a `disPoint()` function defined outside the class, which takes two `Point` objects, `p1` and `p2`, as arguments. This function calculates the squared Euclidean distance between the two points using their coordinates.

Finally, the `disPoint()` function is called with `p1` and `p2` as arguments, and the result is stored in the `res` variable. The square root of `res` is then calculated using the `sqrt()` function from the `<cmath>` library and printed as the distance between `p1` and `p2`.

In summary, the parameterized constructor in the code allows objects of the `Point` class to be created with specified initial values for `x` and `y`. It provides a convenient way to initialize object data members during object creation.
Your point  is (2 , 3)
Your point  is (6 , 7)
The distance between p1 and p2 is 5.65685
